### DISCUSSION TITLE
Return accessToken and refreshToken in Authenticator session cookie

### DISCUSSION THREAD

**Discussion by dan-cooke** [sergiodxa/remix-auth#154] [Q&A] [answered]
Is there a recommended pattern for returning both the access token and refresh token in the session cookie?

Both of these tokens will result in an ecnoded size over 4kb which is the maximum cookie size for most browsers.

At the moment I am thinking about storing the refresh token in a seperate cookie.

**Comment by sergiodxa** [comment, marked as answer]
I would store them in the DB then, also the size of those tokens depends on the provider I imagine, the access token on GitHub for example is a short string.

if your tokens are too big to use a single cookie, you could call authenticate without a success redirect, get the tokens and store them on individual cookies as you said.
